### PR TITLE
Use batches to send sync commands

### LINKED ISSUE #4813: Switch sync fails if a lot of rules need to be installed
**Steps to reproduce:**

1. Create ~400 flows of some switch
2. Manually delete all rules and meters from the switch (deleting of meters is needed to check switch sync on step 3 because rule install commands depends on meter install commands)
3. sync the switch

**Expected result:**
all rules and meters were installed

**Actual result**
timeout

### PR DESCRIPTION
**Pull request by niksv**
Closes #4813

### CODE REVIEW

**Review by dpoltavets — APPROVED** [review]

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java [hunk 1]
```diff
@@ -532,46 +540,59 @@ protected void sendRemoveCommands(SwitchSyncState from, SwitchSyncState to,
...
 
     private List<OfCommand> cleanupDependenciesAndBuildCommands(List<SpeakerData> source) {
         Set<UUID> ids = source.stream()
                 .map(SpeakerData::getUuid)
                 .collect(Collectors.toSet());
         source.forEach(speakerData -> speakerData.getDependsOn().retainAll(ids));
+
+        // As we send commands by several batches we need to sort the list to do not send batch with command A before
+        // batch with command B if A depends on B.
+        source = RuleManagerHelper.sortCommandsByDependencies(source);
```
Comment (rozdy) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java:595]: If A depends on B and A will be in first batch and B in second we will probably get some "Dependency not found error" for second batch. I guess we need to cleanup dependencies after batch split.

**Review by rozdy — CHANGES_REQUESTED** [review]

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java (hunk 1 above)
Comment (niksv) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java:595]: I guess you missed "not" in comment. 
```
As we send commands by several batches we need to sort the list to do NOT send batch with command A before
        // batch with command B if A depends on B.
```
If A depends on B we must send bath with B first

Comment (niksv) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java]: the problem was fixed

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java [hunk 2]
```diff
@@ -532,49 +546,83 @@ protected void sendRemoveCommands(SwitchSyncState from, SwitchSyncState to,
...
                 .map(SpeakerData::getUuid)
                 .collect(Collectors.toSet());
         source.forEach(speakerData -> speakerData.getDependsOn().retainAll(ids));
-        return source.stream()
-                .map(this::toCommand)
-                .collect(Collectors.toList());
+
+        // As we send commands by several batches we need to put command and its dependent commands into same batch
+        List<List<SpeakerData>> groupedCommands = RuleManagerHelper.groupCommandsByDependenciesAndSort(source);
+        List<List<OfCommand>> bathes = new ArrayList<>();
```
Comment (rozdy) [inline src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java:611]: bathes -> batches

**Review by rozdy — APPROVED** [review]

File: src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java (hunk 2 above)
Comment (niksv) [reply src-java/swmanager-topology/swmanager-storm-topology/src/main/java/org/openkilda/wfm/topology/switchmanager/fsm/SwitchSyncFsm.java]: fixed

**Pull request merged**